node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/containerdata/ContainerData.java

```diff
@@ -0,0 +1,90 @@
+package com.yahoo.vespa.hosted.node.admin.containerdata;
+
+import com.yahoo.io.IOUtils;
+import com.yahoo.log.LogLevel;
+import com.yahoo.text.Utf8;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.List;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+/**
+ * This class can be used for adding files that should be accessible in a container's file system after
+ * the container has been started.
+ * <p>
+ * Files that are added will be copied into the destination path on the host.
+ * The entry point for a Docker image will take care of copying
+ * everything into it's right place (ATM only done for config server nodes)
+ * Note: Creating a new instance of this will cleanup all old data in the destination path
+ */
+public class ContainerData {
+
+    private static final Logger log = Logger.getLogger(ContainerData.class.getName());
+    public static final Path containerDataPath = Paths.get("/home/y/var/container-data");
+
+    private final Path destinationPathOnHost;
+
+    public ContainerData(Path destinationPathOnHost) {
+        this.destinationPathOnHost = destinationPathOnHost;
+        try {
+            cleanup(destinationPathOnHost);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void addFile(Path pathInContainer, String data) {
+        Path path = destinationPathOnHost.resolve(pathInContainer);
```

Comment:
This assumes pathInContainer is relative to root - add a check for that and make it clearer from the method/parameter name?